dss/events/handlers/index.py

```diff
@@ -126,25 +132,110 @@ def add_index_data_to_elasticsearch(bundle_key, index_data, logger) -> None:
+                                              refresh=True)
     except Exception as ex:
         logger.error(f"Document not indexed. Exception: {ex}  Index data: {json.dumps(index_data, indent=4)}")
+        raise
+
+
+def find_matching_subscriptions(index_data, logger):
+    percolate_document = {
+        "query": {
+            "percolate": {
+                "field": "query",
+                "document_type": DSS_ELASTICSEARCH_DOC_TYPE,
+                "document": index_data
+            }
+        }
+    }
+    response = ElasticsearchClient.get(logger).search(
+        index=DSS_ELASTICSEARCH_INDEX_NAME,
+        body=percolate_document)
+    logger.debug("Found matching subscription count: %i", len(response['hits']['hits']))
+    subscription_ids = set()
+    for hit in response['hits']['hits']:
+        subscription_ids.add(hit["_id"])
+    return subscription_ids
+
+
+def process_notifications(bundle_id, subscription_ids, logger):
+    for subscription_id in subscription_ids:
+        try:
+            # TODO Batch this request
+            subscription = get_subscription(subscription_id, logger)
+            notify(subscription, bundle_id, logger)
+        except Exception as e:
+            logger.error(f"Error occurred while processing subscription {subscription_id} for bundle {bundle_id}. {e}")
+
+
+def get_subscription(subscription_id, logger):
+    subscription_query = {
+        "query": {
+            "ids": {
+                "type": DSS_ELASTICSEARCH_SUBSCRIPTION_TYPE,
+                "values": [subscription_id]
+            }
+        }
+    }
+    response = ElasticsearchClient.get(logger).search(
+        index=DSS_ELASTICSEARCH_SUBSCRIPTION_INDEX_NAME,
+        body=subscription_query)
+    if len(response['hits']['hits']) == 1:
+        return response['hits']['hits'][0]['_source']
+
+
+def notify(subscription, bundle_id, logger):
+    bundle_uuid, bundle_version = split_bundle_id(bundle_id)
+    transaction_id = uuid.uuid4()
+    payload = {
+        "transaction_id": transaction_id,
+        "subscription_id": subscription['subscription_id'],
+        "query": subscription['query'],
+        "match": {
+            "bundle_uuid": bundle_uuid,
+            "bundle_version": bundle_version
+        }
+    }
+    callback_url = subscription['callback_url']
+    response = requests.post(callback_url, data=payload)
+    if response.status_code == requests.codes.ok:
+        logger.info(f"Successfully notified for subscription {subscription['subscription_id']}"
+                    f" for bundle {bundle_id} with transaction id {transaction_id} Code: {response.status.code}")
+    else:
+        logger.warning(f"Failed notification for subscription {subscription['subscription_id']}"
+                       f" for bundle {bundle_id} with transaction id {transaction_id} Code: {response.status.code}")
+
+
+def split_bundle_id(bundle_id):
+    return bundle_id[:36], bundle_id[37:]
```

Comment:
Changed to: 
`def split_bundle_id(bundle_id):
    split_index = bundle_id.find('.')
    if split_index >= 0:
        return bundle_id[:split_index], bundle_id[(split_index + 1):]
    else:
        return bundle_id, None`